Minimal change for adapting to SocketMode from RTM

ryo-matsuzaki-mdol opened the pull request:
# Whats
https://github.com/rosylilly/ruboty-slack_rtm をフォークしてSlackのSocketModeに対応します。
ruboty を Classic App (Deprecated) でない通常の SlackApp として動かすことができるようになります。

Fork元からの変更点は下記の通りです。

- Slack API のライブラリを https://github.com/slack-ruby/slack-ruby-client に変更
  - Classic App でしか使えない類のAPI呼び出しを変更・廃止
- WebSocket 接続開始用のAPI・トークン変更
- Slack から WebSocket で受け取るデータの構造変化に適応

## How to confirm
`sample` ディレクトリにあるファイル群を使って Slack App を試すことができます。
詳細は README.md をご覧ください。

https://github.com/rosylilly/ruboty-slack_rtm が実装している機能を一通り試したかったので、 `attachment` `file` の2コマンドをサンプルのアプリに追加しています。それぞれ、下の画像のように動きます。
<img width="281" alt="image" src="https://user-images.githubusercontent.com/98291974/163510908-474b1a35-daee-4e7c-878f-5ab0b891d95b.png"><img width="279" alt="image" src="https://user-images.githubusercontent.com/98291974/163510937-dca1d60c-f9ec-4c8f-b584-ee6df890d777.png">

ryo-matsuzaki-mdol committed:
init repositoy

ryo-matsuzaki-mdol committed:
rename some modules from base repository

ryo-matsuzaki-mdol committed:
update slack APIs

- change some API names
- remove deprecated API call
- update websocket connection to SocketMode from RTM
- update data structure handling

ryo-matsuzaki-mdol committed:
add sample SlackApp

takaishi reviewed (commented):
ruboty-slack_rtm のフォークということなので、ruboty-slack_rtm のコードかPRで手を入れたコードなのか判別できるようコミットを分けてもらってもよいでしょうか？

ryo-matsuzaki-mdol commented:
すみません。おそらく根っこからコミットを弄った影響でPRをcloseしてしまいました・・
Reopen ができないようなので、PRは作成しなおします。

ryo-matsuzaki-mdol committed:
add license copyright

ryo-matsuzaki-mdol committed:
remove non-valuable Thread and method alias

ryo-matsuzaki-mdol committed:
re-add method alias for direct message handling

ryo-matsuzaki-mdol committed:
write cases for all slack data types and comments

ryo-matsuzaki-mdol committed:
adapt to review comments

- update ruby version in dev container
- add warnings when received unsupported data or events
- add property methods for ENV reference
- add ENV descriptions in README.md
- make sample env file not hidden
- fix some typo

ryo-matsuzaki-mdol committed:
remove sleep before reconnecting
